<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd"><html xmlns="http://www.w3.org/1999/xhtml"><head>
<title>src/java/security/cert/CertPathValidatorException.java</title>
<meta name="robots" content="noindex,nofollow">
<meta name="generator" content="GLOBAL-6.6.3">
<meta http-equiv="Content-Style-Type" content="text/css">
<link rel="stylesheet" type="text/css" href="../style.css">
</head>
<body>
<a id="TOP" name="TOP"></a><div class="comment">
/* [&lt;][&gt;]<a href="#L200">[^]</a><a href="#L231">[v]</a>[top]<a href="#BOTTOM">[bottom]</a><a href="../mains.html">[index]</a><a href="../help.html">[help]</a> */</div>
<hr>
<h2 class="header">DEFINITIONS</h2>
This source file includes following definitions.
<ol>
<li><a href="#L200" title="Defined at 200.">getCertPath</a></li>
<li><a href="#L212" title="Defined at 212.">getIndex</a></li>
<li><a href="#L227" title="Defined at 227.">getReason</a></li>
<li><a href="#L231" title="Defined at 231.">readObject</a></li>
</ol>
<hr>
<pre>
<div class="comment">
  Copyright (c) 2000, 2013, Oracle and/or its affiliates. All rights reserved.
  ORACLE PROPRIETARY/CONFIDENTIAL. Use is subject to license terms.</div>
<a id="L25" name="L25"></a>  25 
<a id="L26" name="L26"></a>  26 <strong class="reserved">package</strong> <a href="../S/3625.html#L43" title="Defined at 43 in src/javax/management/remote/rmi/RMIConnectionImpl_Stub.java.">java</a>.security.cert;
<a id="L27" name="L27"></a>  27 
<a id="L28" name="L28"></a>  28 <strong class="reserved">import</strong> java.io.InvalidObjectException;
<a id="L29" name="L29"></a>  29 <strong class="reserved">import</strong> java.io.IOException;
<a id="L30" name="L30"></a>  30 <strong class="reserved">import</strong> java.io.ObjectInputStream;
<a id="L31" name="L31"></a>  31 <strong class="reserved">import</strong> java.security.GeneralSecurityException;
<a id="L32" name="L32"></a>  32 
<div class="comment">
  An exception indicating one of a variety of problems encountered when
  validating a certification path.
  <p>
  A <code>CertPathValidatorException</code> provides support for wrapping
  exceptions. The {@link #getCause getCause} method returns the throwable,
  if any, that caused this exception to be thrown.
  </p><p>
  A <code>CertPathValidatorException</code> may also include the
  certification path that was being validated when the exception was thrown,
  the index of the certificate in the certification path that caused the
  exception to be thrown, and the reason that caused the failure. Use the
  {@link #getCertPath getCertPath}, {@link #getIndex getIndex}, and
  {@link #getReason getReason} methods to retrieve this information.
  </p><p>
  <b>Concurrent Access</b>
  </p><p>
  Unless otherwise specified, the methods defined in this class are not
  thread-safe. Multiple threads that need to access a single
  object concurrently should synchronize amongst themselves and
  provide the necessary locking. Multiple threads each manipulating
  separate objects need not synchronize.
  @see CertPathValidator
  @since       1.4
  @author      Yassir Elley</p></div>
<a id="L62" name="L62"></a>  62 <strong class="reserved">public</strong> <strong class="reserved">class</strong> <a href="../R/1219.html" title="Multiple referred from 9 places.">CertPathValidatorException</a> <strong class="reserved">extends</strong> <a href="../S/1369.html#L36" title="Defined at 36 in src/java/security/GeneralSecurityException.java.">GeneralSecurityException</a> <em class="brace">{</em>
<a id="L63" name="L63"></a>  63 
<a id="L64" name="L64"></a>  64     <strong class="reserved">private</strong> <strong class="reserved">static</strong> <strong class="reserved">final</strong> <strong class="reserved">long</strong> serialVersionUID = -3083180014971893139L;
<a id="L65" name="L65"></a>  65 
<div class="comment">
      @serial the index of the certificate in the certification path
      that caused the exception to be thrown</div>
<a id="L70" name="L70"></a>  70     <strong class="reserved">private</strong> <strong class="reserved">int</strong> <a href="../D/24461.html" title="Multiple defined in 6 places.">index</a> = -1;
<a id="L71" name="L71"></a>  71 
<div class="comment">
      @serial the <code>CertPath</code> that was being validated when
      the exception was thrown</div>
<a id="L76" name="L76"></a>  76     <strong class="reserved">private</strong> <a href="../S/1316.html#L120" title="Defined at 120 in src/java/security/cert/CertPath.java.">CertPath</a> certPath;
<a id="L77" name="L77"></a>  77 
<div class="comment">
      @serial the reason the validation failed</div>
<a id="L81" name="L81"></a>  81     <strong class="reserved">private</strong> <a href="../S/1270.html#L251" title="Defined at 251 in src/java/security/cert/CertPathValidatorException.java.">Reason</a> reason = <a href="../S/1270.html#L260" title="Defined at 260 in src/java/security/cert/CertPathValidatorException.java.">BasicReason</a>.UNSPECIFIED;
<a id="L82" name="L82"></a>  82 
<div class="comment">
      Creates a <code>CertPathValidatorException</code> with
      no detail message.</div>
<a id="L87" name="L87"></a>  87     <strong class="reserved">public</strong> CertPathValidatorException() <em class="brace">{</em>
<a id="L88" name="L88"></a>  88         <strong class="reserved">this</strong>(<strong class="reserved">null</strong>, <strong class="reserved">null</strong>);
<a id="L89" name="L89"></a>  89     <em class="brace">}</em>
<a id="L90" name="L90"></a>  90 
<div class="comment">
      Creates a <code>CertPathValidatorException</code> with the given
      detail message. A detail message is a <code>String</code> that
      describes this particular exception.
      @param msg the detail message</div>
<a id="L98" name="L98"></a>  98     <strong class="reserved">public</strong> CertPathValidatorException(<a href="../D/8419.html" title="Multiple defined in 2 places.">String</a> <a href="../S/6957.html#L1026" title="Defined at 1026 in src/com/sun/corba/se/impl/io/ObjectStreamClass.java.">msg</a>) <em class="brace">{</em>
<a id="L99" name="L99"></a>  99         <strong class="reserved">this</strong>(<a href="../S/6957.html#L1026" title="Defined at 1026 in src/com/sun/corba/se/impl/io/ObjectStreamClass.java.">msg</a>, <strong class="reserved">null</strong>);
<a id="L100" name="L100"></a> 100     <em class="brace">}</em>
<a id="L101" name="L101"></a> 101 
<div class="comment">
      Creates a <code>CertPathValidatorException</code> that wraps the
      specified throwable. This allows any exception to be converted into a
      <code>CertPathValidatorException</code>, while retaining information
      about the wrapped exception, which may be useful for debugging. The
      detail message is set to (<code>cause==null ? null : cause.toString()</code>)
      (which typically contains the class and detail message of
      cause).
      @param cause the cause (which is saved for later retrieval by the
      {@link #getCause getCause()} method). (A <code>null</code> value is
      permitted, and indicates that the cause is nonexistent or unknown.)</div>
<a id="L115" name="L115"></a> 115     <strong class="reserved">public</strong> CertPathValidatorException(<a href="../S/1726.html#L114" title="Defined at 114 in src/java/lang/Throwable.java.">Throwable</a> cause) <em class="brace">{</em>
<a id="L116" name="L116"></a> 116         <strong class="reserved">this</strong>((cause == <strong class="reserved">null</strong> ? <strong class="reserved">null</strong> : cause.<a href="../D/35493.html" title="Multiple defined in 1021 places.">toString</a>()), cause);
<a id="L117" name="L117"></a> 117     <em class="brace">}</em>
<a id="L118" name="L118"></a> 118 
<div class="comment">
      Creates a <code>CertPathValidatorException</code> with the specified
      detail message and cause.
      @param msg the detail message
      @param cause the cause (which is saved for later retrieval by the
      {@link #getCause getCause()} method). (A <code>null</code> value is
      permitted, and indicates that the cause is nonexistent or unknown.)</div>
<a id="L128" name="L128"></a> 128     <strong class="reserved">public</strong> CertPathValidatorException(<a href="../D/8419.html" title="Multiple defined in 2 places.">String</a> <a href="../S/6957.html#L1026" title="Defined at 1026 in src/com/sun/corba/se/impl/io/ObjectStreamClass.java.">msg</a>, <a href="../S/1726.html#L114" title="Defined at 114 in src/java/lang/Throwable.java.">Throwable</a> cause) <em class="brace">{</em>
<a id="L129" name="L129"></a> 129         <strong class="reserved">this</strong>(<a href="../S/6957.html#L1026" title="Defined at 1026 in src/com/sun/corba/se/impl/io/ObjectStreamClass.java.">msg</a>, cause, <strong class="reserved">null</strong>, -1);
<a id="L130" name="L130"></a> 130     <em class="brace">}</em>
<a id="L131" name="L131"></a> 131 
<div class="comment">
      Creates a <code>CertPathValidatorException</code> with the specified
      detail message, cause, certification path, and index.
      @param msg the detail message (or <code>null</code> if none)
      @param cause the cause (or <code>null</code> if none)
      @param certPath the certification path that was in the process of
      being validated when the error was encountered
      @param index the index of the certificate in the certification path
      that caused the error (or -1 if not applicable). Note that
      the list of certificates in a <code>CertPath</code> is zero based.
      @throws IndexOutOfBoundsException if the index is out of range
      {@code (index &lt; -1 || (certPath != null &amp;&amp; index &gt;=
      certPath.getCertificates().size()) }
      @throws IllegalArgumentException if <code>certPath</code> is
      <code>null</code> and <code>index</code> is not -1</div>
<a id="L149" name="L149"></a> 149     <strong class="reserved">public</strong> CertPathValidatorException(<a href="../D/8419.html" title="Multiple defined in 2 places.">String</a> <a href="../S/6957.html#L1026" title="Defined at 1026 in src/com/sun/corba/se/impl/io/ObjectStreamClass.java.">msg</a>, <a href="../S/1726.html#L114" title="Defined at 114 in src/java/lang/Throwable.java.">Throwable</a> cause,
<a id="L150" name="L150"></a> 150             <a href="../S/1316.html#L120" title="Defined at 120 in src/java/security/cert/CertPath.java.">CertPath</a> certPath, <strong class="reserved">int</strong> <a href="../D/24461.html" title="Multiple defined in 6 places.">index</a>) <em class="brace">{</em>
<a id="L151" name="L151"></a> 151         <strong class="reserved">this</strong>(<a href="../S/6957.html#L1026" title="Defined at 1026 in src/com/sun/corba/se/impl/io/ObjectStreamClass.java.">msg</a>, cause, certPath, <a href="../D/24461.html" title="Multiple defined in 6 places.">index</a>, <a href="../S/1270.html#L260" title="Defined at 260 in src/java/security/cert/CertPathValidatorException.java.">BasicReason</a>.UNSPECIFIED);
<a id="L152" name="L152"></a> 152     <em class="brace">}</em>
<a id="L153" name="L153"></a> 153 
<div class="comment">
      Creates a <code>CertPathValidatorException</code> with the specified
      detail message, cause, certification path, index, and reason.
      @param msg the detail message (or <code>null</code> if none)
      @param cause the cause (or <code>null</code> if none)
      @param certPath the certification path that was in the process of
      being validated when the error was encountered
      @param index the index of the certificate in the certification path
      that caused the error (or -1 if not applicable). Note that
      the list of certificates in a <code>CertPath</code> is zero based.
      @param reason the reason the validation failed
      @throws IndexOutOfBoundsException if the index is out of range
      {@code (index &lt; -1 || (certPath != null &amp;&amp; index &gt;=
      certPath.getCertificates().size()) }
      @throws IllegalArgumentException if <code>certPath</code> is
      <code>null</code> and <code>index</code> is not -1
      @throws NullPointerException if <code>reason</code> is <code>null</code>
      @since 1.7</div>
<a id="L175" name="L175"></a> 175     <strong class="reserved">public</strong> CertPathValidatorException(<a href="../D/8419.html" title="Multiple defined in 2 places.">String</a> <a href="../S/6957.html#L1026" title="Defined at 1026 in src/com/sun/corba/se/impl/io/ObjectStreamClass.java.">msg</a>, <a href="../S/1726.html#L114" title="Defined at 114 in src/java/lang/Throwable.java.">Throwable</a> cause,
<a id="L176" name="L176"></a> 176             <a href="../S/1316.html#L120" title="Defined at 120 in src/java/security/cert/CertPath.java.">CertPath</a> certPath, <strong class="reserved">int</strong> <a href="../D/24461.html" title="Multiple defined in 6 places.">index</a>, <a href="../S/1270.html#L251" title="Defined at 251 in src/java/security/cert/CertPathValidatorException.java.">Reason</a> reason) <em class="brace">{</em>
<a id="L177" name="L177"></a> 177         <strong class="reserved">super</strong>(<a href="../S/6957.html#L1026" title="Defined at 1026 in src/com/sun/corba/se/impl/io/ObjectStreamClass.java.">msg</a>, cause);
<a id="L178" name="L178"></a> 178         <strong class="reserved">if</strong> (certPath == <strong class="reserved">null</strong> &amp;&amp; <a href="../D/24461.html" title="Multiple defined in 6 places.">index</a> != -1) <em class="brace">{</em>
<a id="L179" name="L179"></a> 179             <strong class="reserved">throw</strong> <strong class="reserved">new</strong> <a href="../D/4055.html" title="Multiple defined in 2 places.">IllegalArgumentException</a>();
<a id="L180" name="L180"></a> 180         <em class="brace">}</em>
<a id="L181" name="L181"></a> 181         <strong class="reserved">if</strong> (<a href="../D/24461.html" title="Multiple defined in 6 places.">index</a> &lt; -1 ||
<a id="L182" name="L182"></a> 182             (certPath != <strong class="reserved">null</strong> &amp;&amp; <a href="../D/24461.html" title="Multiple defined in 6 places.">index</a> &gt;= certPath.<a href="../D/17559.html" title="Multiple defined in 6 places.">getCertificates</a>().<a href="../D/34593.html" title="Multiple defined in 204 places.">size</a>())) <em class="brace">{</em>
<a id="L183" name="L183"></a> 183             <strong class="reserved">throw</strong> <strong class="reserved">new</strong> <a href="../S/1778.html#L38" title="Defined at 38 in src/java/lang/IndexOutOfBoundsException.java.">IndexOutOfBoundsException</a>();
<a id="L184" name="L184"></a> 184         <em class="brace">}</em>
<a id="L185" name="L185"></a> 185         <strong class="reserved">if</strong> (reason == <strong class="reserved">null</strong>) <em class="brace">{</em>
<a id="L186" name="L186"></a> 186             <strong class="reserved">throw</strong> <strong class="reserved">new</strong> <a href="../S/1559.html#L53" title="Defined at 53 in src/java/lang/NullPointerException.java.">NullPointerException</a>("reason can't be null");
<a id="L187" name="L187"></a> 187         <em class="brace">}</em>
<a id="L188" name="L188"></a> 188         <strong class="reserved">this</strong>.certPath = certPath;
<a id="L189" name="L189"></a> 189         <strong class="reserved">this</strong>.<a href="../D/24461.html" title="Multiple defined in 6 places.">index</a> = <a href="../D/24461.html" title="Multiple defined in 6 places.">index</a>;
<a id="L190" name="L190"></a> 190         <strong class="reserved">this</strong>.reason = reason;
<a id="L191" name="L191"></a> 191     <em class="brace">}</em>
<a id="L192" name="L192"></a> 192 
<div class="comment">
      Returns the certification path that was being validated when
      the exception was thrown.
      @return the <code>CertPath</code> that was being validated when
      the exception was thrown (or <code>null</code> if not specified)</div>
<a id="L200" name="L200"></a> 200     <strong class="reserved">public</strong> <a href="../S/1316.html#L120" title="Defined at 120 in src/java/security/cert/CertPath.java.">CertPath</a> getCertPath() <em class="brace">{</em>
<a id="L201" name="L201"></a> 201         <strong class="reserved">return</strong> <strong class="reserved">this</strong>.certPath;
<a id="L202" name="L202"></a> 202     <em class="brace">}</em>
<a id="L203" name="L203"></a> 203 
<div class="comment">
      Returns the index of the certificate in the certification path
      that caused the exception to be thrown. Note that the list of
      certificates in a <code>CertPath</code> is zero based. If no
      index has been set, -1 is returned.
      @return the index that has been set, or -1 if none has been set</div>
<a id="L212" name="L212"></a> 212     <strong class="reserved">public</strong> <strong class="reserved">int</strong> <a href="../R/17035.html" title="Multiple referred from 302 places.">getIndex</a>() <em class="brace">{</em>
<a id="L213" name="L213"></a> 213         <strong class="reserved">return</strong> <strong class="reserved">this</strong>.<a href="../D/24461.html" title="Multiple defined in 6 places.">index</a>;
<a id="L214" name="L214"></a> 214     <em class="brace">}</em>
<a id="L215" name="L215"></a> 215 
<div class="comment">
      Returns the reason that the validation failed. The reason is
      associated with the index of the certificate returned by
      {@link #getIndex}.
      @return the reason that the validation failed, or
         <code>BasicReason.UNSPECIFIED</code> if a reason has not been
         specified
      @since 1.7</div>
<a id="L227" name="L227"></a> 227     <strong class="reserved">public</strong> <a href="../S/1270.html#L251" title="Defined at 251 in src/java/security/cert/CertPathValidatorException.java.">Reason</a> <a href="../R/18619.html" title="Multiple referred from 5 places.">getReason</a>() <em class="brace">{</em>
<a id="L228" name="L228"></a> 228         <strong class="reserved">return</strong> <strong class="reserved">this</strong>.reason;
<a id="L229" name="L229"></a> 229     <em class="brace">}</em>
<a id="L230" name="L230"></a> 230 
<a id="L231" name="L231"></a> 231     <strong class="reserved">private</strong> <strong class="reserved">void</strong> <a href="../R/25518.html" title="Multiple referred from 189 places.">readObject</a>(<a href="../S/1378.html#L206" title="Defined at 206 in src/java/io/ObjectInputStream.java.">ObjectInputStream</a> <a href="../D/34926.html" title="Multiple defined in 21 places.">stream</a>)
<a id="L232" name="L232"></a> 232         <strong class="reserved">throws</strong> <a href="../S/1705.html#L53" title="Defined at 53 in src/java/lang/ClassNotFoundException.java.">ClassNotFoundException</a>, <a href="../S/1392.html#L39" title="Defined at 39 in src/java/io/IOException.java.">IOException</a> <em class="brace">{</em>
<a id="L233" name="L233"></a> 233         <a href="../D/34926.html" title="Multiple defined in 21 places.">stream</a>.<a href="../D/14746.html" title="Multiple defined in 2 places.">defaultReadObject</a>();
<a id="L234" name="L234"></a> 234         <strong class="reserved">if</strong> (reason == <strong class="reserved">null</strong>) <em class="brace">{</em>
<a id="L235" name="L235"></a> 235             reason = <a href="../S/1270.html#L260" title="Defined at 260 in src/java/security/cert/CertPathValidatorException.java.">BasicReason</a>.UNSPECIFIED;
<a id="L236" name="L236"></a> 236         <em class="brace">}</em>
<a id="L237" name="L237"></a> 237         <strong class="reserved">if</strong> (certPath == <strong class="reserved">null</strong> &amp;&amp; <a href="../D/24461.html" title="Multiple defined in 6 places.">index</a> != -1) <em class="brace">{</em>
<a id="L238" name="L238"></a> 238             <strong class="reserved">throw</strong> <strong class="reserved">new</strong> <a href="../S/1428.html#L38" title="Defined at 38 in src/java/io/InvalidObjectException.java.">InvalidObjectException</a>("certpath is null and index != -1");
<a id="L239" name="L239"></a> 239         <em class="brace">}</em>
<a id="L240" name="L240"></a> 240         <strong class="reserved">if</strong> (<a href="../D/24461.html" title="Multiple defined in 6 places.">index</a> &lt; -1 ||
<a id="L241" name="L241"></a> 241             (certPath != <strong class="reserved">null</strong> &amp;&amp; <a href="../D/24461.html" title="Multiple defined in 6 places.">index</a> &gt;= certPath.<a href="../D/17559.html" title="Multiple defined in 6 places.">getCertificates</a>().<a href="../D/34593.html" title="Multiple defined in 204 places.">size</a>())) <em class="brace">{</em>
<a id="L242" name="L242"></a> 242             <strong class="reserved">throw</strong> <strong class="reserved">new</strong> <a href="../S/1428.html#L38" title="Defined at 38 in src/java/io/InvalidObjectException.java.">InvalidObjectException</a>("index out of range");
<a id="L243" name="L243"></a> 243         <em class="brace">}</em>
<a id="L244" name="L244"></a> 244     <em class="brace">}</em>
<a id="L245" name="L245"></a> 245 
<div class="comment">
      The reason the validation algorithm failed.
      @since 1.7</div>
<a id="L251" name="L251"></a> 251     <strong class="reserved">public</strong> <strong class="reserved">static</strong> <strong class="reserved">interface</strong> <a href="../R/6509.html" title="Multiple referred from 5 places.">Reason</a> <strong class="reserved">extends</strong> <a href="../S/3625.html#L43" title="Defined at 43 in src/javax/management/remote/rmi/RMIConnectionImpl_Stub.java.">java</a>.io.<a href="../S/1407.html#L169" title="Defined at 169 in src/java/io/Serializable.java.">Serializable</a> <em class="brace">{</em> <em class="brace">}</em>
<a id="L252" name="L252"></a> 252 
<a id="L253" name="L253"></a> 253 
<div class="comment">
      The BasicReason enumerates the potential reasons that a certification
      path of any type may be invalid.
      @since 1.7</div>
<a id="L260" name="L260"></a> 260     <strong class="reserved">public</strong> <strong class="reserved">static</strong> <strong class="reserved">enum</strong> <a href="../R/716.html" title="Multiple referred from 3 places.">BasicReason</a> <strong class="reserved">implements</strong> <a href="../S/1270.html#L251" title="Defined at 251 in src/java/security/cert/CertPathValidatorException.java.">Reason</a> <em class="brace">{</em>
<div class="comment">
          Unspecified reason.</div>
<a id="L264" name="L264"></a> 264         UNSPECIFIED,
<a id="L265" name="L265"></a> 265 
<div class="comment">
          The certificate is expired.</div>
<a id="L269" name="L269"></a> 269         EXPIRED,
<a id="L270" name="L270"></a> 270 
<div class="comment">
          The certificate is not yet valid.</div>
<a id="L274" name="L274"></a> 274         NOT_YET_VALID,
<a id="L275" name="L275"></a> 275 
<div class="comment">
          The certificate is revoked.</div>
<a id="L279" name="L279"></a> 279         REVOKED,
<a id="L280" name="L280"></a> 280 
<div class="comment">
          The revocation status of the certificate could not be determined.</div>
<a id="L284" name="L284"></a> 284         UNDETERMINED_REVOCATION_STATUS,
<a id="L285" name="L285"></a> 285 
<div class="comment">
          The signature is invalid.</div>
<a id="L289" name="L289"></a> 289         INVALID_SIGNATURE,
<a id="L290" name="L290"></a> 290 
<div class="comment">
          The public key or the signature algorithm has been constrained.</div>
<a id="L294" name="L294"></a> 294         ALGORITHM_CONSTRAINED
<a id="L295" name="L295"></a> 295     <em class="brace">}</em>
<a id="L296" name="L296"></a> 296 <em class="brace">}</em>
</pre>
<hr>
<div class="comment">
/* [&lt;][&gt;]<a href="#L200">[^]</a><a href="#L231">[v]</a><a href="#TOP">[top]</a>[bottom]<a href="../mains.html">[index]</a><a href="../help.html">[help]</a> */</div>
</body></html>